daffodil-core/src/main/scala/org/apache/daffodil/grammar/LocalElementGrammarMixin.scala

```diff
@@ -211,7 +214,7 @@ trait LocalElementGrammarMixin extends GrammarMixin { self: ElementBase =>
       case (TrailingStr, Implicit__, UNB, ___) => separatedContentWithMinUnboundedWithoutTrailingEmpties // we're depending on optionalEmptyPart failing on empty content.
       case (TrailingStr, Implicit__, max, ___) => separatedContentAtMostNWithoutTrailingEmpties
       case (Always_____, Implicit__, UNB, ___) => separatedContentWithMinUnbounded
-      case (Always_____, Implicit__, max, ___) => separatedContentAtMostN
+      case (Always_____, Implicit__, max, ___) => separatedContentAtMostNWithoutTrailingEmpties
```

Comment:
I am not 100% sure that this is correct behaviour, but it results in the same behaviour as before these changes.  This prevents TestArrayOptionalElem.test_backtrack1Text from failing.